### PR TITLE
added merge_maf part

### PR DESCRIPTION
**Pull request by svural**

### CODE REVIEW

**Comment by allanbolipata** [comment]
add to whitelist

**Comment by stevekm** [comment]
yeah I think you need to do the merge between the Facets maf & the portal maf

File: tests/test_update_cBioPortal_data.cwl.py [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+        # maf_lines = [
+        #     ['# comment 1'],
+        #     ['# comment 2'],
+        #     ['Hugo_Symbol', 't_depth', 't_alt_count','tcn','lcn','expected_alt_copies','ccf_expected_copies','ccf_expected_copies_lower','ccf_expected_copies_upper'],
+        #     ['SUFU', '100', '75','4','0.127','0.615','0.375'],
+        #     ['GOT1', '100', '1' ,'4','0.127','0.615','0.375'], # need to change the values
+        #     ['SOX9', '100', '0' ,'4','0.127','0.615','0.375'],
+        # ]
+
+def setUp(self):
```
Comment (stevekm) [inline tests/test_update_cBioPortal_data.cwl.py:35]: if you are using a setup method for the unit test then it should be one of these;

https://docs.python.org/3/library/unittest.html#unittest.TestCase.setUp

example;

https://github.com/mskcc/dmp2cmo/blob/febf153f06d050334bae661dc51f3b0c8b238a89/dmp2cmo/test_update.py#L46-L47

File: cwl/facets-suite-workflow.cwl [hunk 2]
```diff
@@ -426,6 +426,21 @@ steps:
     out:
       [ output_file, failed_txt, stdout_txt, stderr_txt ]
 
+  merge_maf:
+    run: update_cBioPortal_data.cwl
+    in:
+      pair_id: pair_id
+      subcommand:
+        valueFrom: ${ return "merge_mafs"; }
+      input_file: check_label_maf_normal/output_file
```
Comment (stevekm) [inline cwl/facets-suite-workflow.cwl:435]: yea I think this is supposed to be the portal maf file `data_mutations_extended.txt` file

File: cwl/facets-suite-workflow.cwl [hunk 3]
```diff
@@ -426,6 +426,21 @@ steps:
...
+      output_filename:
+        valueFrom: $(inputs.pair_id)_hisens.ccf.portal.maf
+      facets_txt: check_run_facets/single_facets_txt
+    scatter: [input_file]
+    scatterMethod: dotproduct
```
Comment (stevekm) [inline cwl/facets-suite-workflow.cwl:440]: shouldnt need to scatter this, it should just be 1 portal file vs. 1 facets file, I think

File: tests/test_workflow_with_facets.py [hunk 4]
```diff
@@ -801,6 +801,10 @@ def test_run_worflow_two_mafs(self):
             comments, mutations = load_mutations(os.path.join(output_dir, 'portal', 'data_mutations_extended.txt'))
             self.assertEqual(len(mutations), 27)
 
+            # the clonality column needs to have been added in the workflow output
+            for mut in mutations:
+                self.assertTrue('ASCN.CLONAL' in mut)
```
Comment (stevekm) [inline tests/test_workflow_with_facets.py:806]: @svural this is the test that needs to pass

**Pull request merged**